recipes/proj/all/patches/use-cmake-targets-8.2.0.patch

```diff
@@ -0,0 +1,31 @@
+--- a/src/lib_proj.cmake
++++ b/src/lib_proj.cmake
+@@ -453,8 +453,7 @@ if(USE_THREAD AND Threads_FOUND AND CMAKE_USE_PTHREADS_INIT)
+   target_link_libraries(proj PRIVATE ${CMAKE_THREAD_LIBS_INIT})
+ endif()
+ 
+-target_include_directories(proj PRIVATE ${SQLITE3_INCLUDE_DIR})
+-target_link_libraries(proj PRIVATE ${SQLITE3_LIBRARY})
++target_link_libraries(proj PRIVATE CONAN_PKG::sqlite3)
```

Comment:
This will break in conan 2.
Can we use `Sqlite3::Sqlite3`?

Reply:
`Sqlite3::Sqlite3` doesn't exist. If you refer to `SQLite::SQLite3`, it can't work here without more patch.

The propor fix to submit upstream would be:

- change their `FindSqlite3.cmake` to follow official one (required if consumers are using CMake < 3.14): filename, target name, variables names
- change top CMakeLists to:
  - APPEND `${PROJ_SOURCE_DIR}/cmake` in `CMAKE_MODULE_PATH`, instead of prepend instead of PREPEND: https://github.com/OSGeo/PROJ/blob/01afa55e56148a668eabeb271a15352f920b3bca/CMakeLists.txt#L103
  - change this 2 lines to honor official `FindSQLite3.cmake`: https://github.com/OSGeo/PROJ/blob/01afa55e56148a668eabeb271a15352f920b3bca/CMakeLists.txt#L177-L178
  - use `SQLite::SQLite3` target to link sqlite3 into proj.

For `TIFF::TIFF`, PROJ now require CMake >=3.9, so we can already submit the patch upstream (this target is available since CMake 3.5: https://cmake.org/cmake/help/latest/module/FindTIFF.html)

For `CURL::libcurl`, it requires CMake >= 3.12, so it can't be patched upstream without a bump of CMake min version.

Reply:
Thanks for diving in.

When conan 2.0 happens, will we drop support for all older proj versions?
I count 12 versions in `conandata.yml`.

Patch all versions is not reasonable, imho.

Reply:
Yes, we can keep the last minor.patch version of major versions 6, 7 and 8